roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -33,12 +33,23 @@ class DockerImageAvailability(DockerHostMixin, OpenShiftCheck):
     # we use python-docker-py to check local docker for images, and skopeo
     # to look for images available remotely without waiting to pull them.
     dependencies = ["python-docker-py", "skopeo"]
-    skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false docker://{registry}/{image}"
 
     def __init__(self, *args, **kwargs):
         super(DockerImageAvailability, self).__init__(*args, **kwargs)
         # record whether we could reach a registry or not (and remember results)
         self.reachable_registries = {}
+        self.skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false"
+
+        oreg_auth_user = self.get_var('oreg_auth_user', default='')
+        oreg_auth_password = self.get_var('oreg_auth_password', default='')
+        if oreg_auth_user != '' and oreg_auth_password != '':
+            # skopeo should complete without complaining if credentials are
+            # passed to an unauthenticated registry.
+            self.skopeo_img_check_command += " --creds='{oau}:{oap}'".format(oau=oreg_auth_user, oap=oreg_auth_password)
```

Comment:
This string gets displayed to the user if there's a problem (so that they know what command was run), so this should not be format()ed until it is actually being used (also, what if someone's password was `{foobar}`). Also the names should make sense to a user e.g. `--creds={user}:{pass}`

Also these should be shell-escaped. Single quotes won't help when the password is `foo'bar!!`

Just format it down in `is_available_skopeo_image` where it's used. `format()` doesn't mind some extra named parameters if there are no creds.

Reply:
I can move the format logic below, your example of '{foobar}' for password is a good point.

Shell-escaping is up to the user.  I don't think we should translate user input unless absolutely necessary.

Reply:
Requiring users to be aware of shell-escaping for passwords would only bite us later. However, I forgot that the `command` module doesn't run through the shell; so this not an issue.

Reply:
I'm of two minds on the shell escaping. Do we have examples elsewhere that either requires the user to manually escape OR handles escaping for them?